pkgs/development/tools/build-managers/xmake/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchgit
+# optional list of extra waf tools, e.g. `[ "doxygen" "pytest" ]`
+
+}:
+let
+in
+stdenv.mkDerivation rec {
+  pname = "xmake";
+  version = "2.6.7";
+
+  src = fetchgit {
+    url = "https://github.com/xmake-io/xmake";
```

Comment:
Please use fetchFromGitHub

Reply:
It depends on git submodules, fetchFromGitHub didn't checkout recusive. that's leads to a compile error.

Reply:
> It depends on git submodules, fetchFromGitHub didn't checkout recusive. that's leads to a compile error.

It's possible to fetch submodules with `fetchFromGitHub`, add `fetchSubmodules = true`.